src/composer/src/build.rs

```diff
@@ -248,12 +250,58 @@ fn constructor_serialize_args(
     Ok(args_file_path)
 }
 
+fn comp_gen_make_cmd_scan_lib_constant(constants: &[String]) -> Result<Vec<String>, String> {
+    println!("Scanning libraries for constants: {:?}", constants); // Print the constants being scanned for
+    let constants_str = constants.join(" ");
+    let output = Command::new("make")
+        .arg("-C")
+        .arg("src")
+        .arg("--no-print-directory")
+        .arg("scan_libs_for_constants")
+        .env("CONSTANTS", constants_str)
+        .output()
+        .expect("Failed to execute make command");
+    
+    if !output.status.success() {
+        return Err(String::from_utf8_lossy(&output.stderr).to_string());
+    }
+
+    let rebuild_libs_file = File::open("src/components/lib/rebuild_libs.txt").expect("Failed to open rebuild_libs.txt");
```

Comment:
We discussed this previously, but I thought I'd write it here for the record. We certainly do not want to use files to store values that should be passed through data-structures. If you want some input on designing this, I'm happy to help.

Again, I very much could be missing something here, as I can see this file is recording information for all libraries.

I'm not sure if the composer can be executed validly from another directly other than the Composite source root. If so, this hard-coded relative directory is not appropriate. I honestly don't remember if that is an assumption we make elsewhere.